torchao/quantization/autoquant.py

```diff
@@ -513,3 +513,42 @@ def clean_up_autoquant_hooks_and_attrs():
         model(*example_input)
 
     return model
+
+class AutoQuantConfig:
```

Comment:
sorry I keep harping on names :P 

But this sounds like it's a Cache instead?

In which case wouldn't something like this be clearer `torch.autoquant(model, cache_path = "test.pkl")` which would call `apply_to_model()` on behalf of the user?

And shouldn't you always `save`?

Reply:
The cache (https://github.com/pytorch/ao/blob/main/torchao/quantization/autoquant.py#L22) is something different. I still need to write automatic serialization for that.

unless you're saying this new thing is reall a Cache? How would you define the difference between a cache and a config?

Reply:
I think of configs as a class representing a common set of input arguments to a function while a cache is something that saves the result of some search process